app/views/subnets/_fields.html.erb

```diff
@@ -1,20 +1,25 @@
 <%= text_f f, :name %>
+<%= selectable_f f, :type,
+  options_for_select(Subnet.types_with_form_data, @subnet.type),
+  {:label => _("Protocol")},
+  :disabled => !f.object.new_record? %>
 <%= text_f f, :network %>
-<%= text_f f, :mask, :help_inline => _("Netmask for this subnet") %>
+<%= text_f f, :cidr, :label => _("CIDR"), :help_inline => _("CIDR for this subnet") %>
```

Comment:
@dLobatog: To be honest I don't want to argue about this. But If I'm not mistaken "prefix length" is an IPv6 term, for IPv4 it's called "suffix". CIDR just says, that you don't specify the subnet mask in dotted decimal notation, but you count the number of 1-bits in the subnet mask. That's why I chose CIDR, but I don't mind changing this.
I guess it would be better to show the subnet mask in dotted decimal notation for IPv4 and the prefix length for IPv6. That would require us to filter the opposite param in the controller. Do you think, that's worth the effort? Is there a better approach?